src/redux/books/books.js

```diff
@@ -59,4 +19,101 @@ export const removeBook = (id) => ({
   payload: { id },
 });
 
+export const fetchBooks = (books) => ({
+  type: FETCH_BOOKS,
+  payload: books,
+});
+
+export const fetchBooksError = (error) => ({
+  type: FETCH_BOOKS_ERROR,
+  payload: error,
+});
+
+export const fetchBooksLoading = () => ({
+  type: FETCH_BOOKS_LOADING,
+});
+
+// Default state
+const books = {
+  books: [],
+  loading: false,
+  error: null,
+};
+
+export const getBooks = () => (dispatch) => {
+  dispatch(fetchBooksLoading());
+  fetch(URL)
+    .then((response) => response.json())
+    .then((data) => {
+      const newBooks = [];
+      Object.keys(data).forEach((key) => {
+        if (key) {
+          newBooks.push({
+            ...data[key][0],
+            item_id: key,
+          });
+        }
+      });
+      dispatch(fetchBooks(newBooks));
+    })
+    .catch((error) => {
+      dispatch(fetchBooksError(error.message));
+    });
+};
+
+export const postBook = (book) => (dispatch) => {
+  fetch(URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify((book)),
+  }).then(() => {
+    dispatch(addBook(book));
+  });
+};
+
+export const deleteBook = (id) => (dispatch) => {
+  fetch(`${URL}/${id}`, {
+    method: 'DELETE',
+  })
+    .then(() => {
+      dispatch(getBooks());
+    });
+};
+
+// Reducer function
+const booksReducer = (state = books, action) => {
+  switch (action.type) {
+    case FETCH_BOOKS_LOADING:
+      return {
+        ...state,
+        loading: true,
+      };
+    case ADD_BOOK:
+      return {
+        ...state,
+        loading: false,
+        books: [...state.books, action.payload],
+      };
+    // case REMOVE_BOOK:
+    //   return deleteBook(action.payload.id);
+    //   // state.filter((book) => book.id !== action.payload.id);
```

Comment:
- [ ] Kindly this case is still important as is the one that will filter the store and remove the deleted book
Example
```js
case REMOVE_BOOK:
    return state.filter((book) => book.id !== action.payload.id);
```

Reply:
Resolved